src/prefect/tasks/azure/blobstorage.py

```diff
@@ -57,10 +57,13 @@ def run(
         # get Azure credentials
         azure_credentials = Secret(azure_credentials_secret).get()
         az_account_name = azure_credentials["ACCOUNT_NAME"]
-        az_account_key = azure_credentials["ACCOUNT_KEY"]
+        az_account_key = azure_credentials.get("ACCOUNT_KEY")
+        az_sas_token = azure_credentials.get("SAS_TOKEN")
 
         blob_service = azure.storage.blob.BlockBlobService(
-            account_name=az_account_name, account_key=az_account_key
+            account_name=az_account_name,
+            account_key=az_account_key,
+            sas_token=az_sas_token,
```

Comment:
What happens if both `account_key` and `sas_token` are provided? Is it rightfully handled or is an error raised?

Reply:
Good question. 

For `Datastore.register_azure_blob_container` it basically comes down to a call to `azureml.data.datastore_client._DatastoreClient._get_credential_type(account_key, sas_token)` which gives preference to `account_key`. No error. 

For `BlockBlobService` we initiate its base class `azure.storage.blob.baseblobservice.BaseBlobService`, which follows the same logic; `account_key` is given preference, and no error. 

This functionality should maybe be mentioned in the docstrings?

Reply:
Yeah! Just add a quick note about preference!

Reply:
Updated!